### ISSUE TITLE
Change algorithm for data slider under map / Изменить алгоритм работы слайдера выбора дат под картой

### ISSUE THREAD

**Issue by paul-k-pastvu** [PastVu/pastvu#217]
EN: When the map data slider is set to 1945-1951, and the photo has the range of 1946-1950 then photo is now included on the minimap, even though the range falls within the filter parameters. This is because only the lowest range value is used when filtering.

Proposal - choosing images as intersection bu two data ranges.

----

RU: Если фильтр дат под картой установлен на 1945-1951, а года фотографии на 1946-1950, то фотография не отображается на миникарте, хотя временной промежуток попадает в параметры заданного фильтра. Проблема в том, что при фильтрации учитывается только нижняя граница временного промежутка фотографии.)

Предложение: отбирать изображения по пересечению двух диапазонов дат.

**Comment by paul-k-pastvu** [comment]
@klimashkin -  я правильно помню, что там была какая-то проблема с кластеризацией, из-за которой так сделать не удалось?

**Comment by klimashkin** [comment]
Right, photos get aggregated into a cluster by one of the years, that allows clusters to be static, otherwise they would need to be computed on each request, which is extremely expensive

**Comment by CraftCoderr** [comment]
As I understand, due to clustering problem we can't show correct count of photos in cluster, but we can show photos itself when zoomed in.
